cmd/dagger/call.go

```diff
@@ -40,6 +42,18 @@ var callCmd = &FuncCommand{
 			}
 		}
 
+		// There's no fields in `Container` that trigger container execution so
+		// we use `sync` first to evaluate, and then load the new `Container`
+		// from that response before continuing.
+		if typeName == Container {
+			var cid dagger.ContainerID
+			fc.Select("sync")
+			if err := fc.Request(ctx, &cid); err != nil {
+				return err
+			}
+			fc.q = fc.q.Root().Select("loadContainerFromID").Arg("id", cid)
```

Comment:
nonblocking, more forward-looking: I think we don't actually need to do this since `sync` will always just return the original ID with DagQL. Not sure if it mattered before, but we also added this before `Void` iirc. I wonder if `sync` should just return `Void` now?

Reply:
Not according to:
- https://github.com/dagger/dagger/issues/7788

Reply:
>  I wonder if `sync` should just return `Void` now?

The original reason for returning `ContainerID` was actually to make it easier to continue chaining and avoid the need for a `Void` at the time:

```diff
- ctr := test()
- _, err := ctr.Sync(ctx)
+ ctr, err := test().Sync(ctx)
if err != nil {
    return "", err
}
return ctr.Stdout(ctx)
```

Doesn't seem that different but see Python:

```diff
- ctr = test()
- await ctr.sync()
- return await ctr.stdout()
+ return await (await test().sync()).stdout()
```

But now that we have `Void`, and if we really don't need to load from ID, considering also that there's https://github.com/dagger/dagger/pull/7773, then the exception here stands out a bit. `sync` is the only one that works like this so it needs special handling in codegen.